src/main/java/ui/activity/Activity.java

```diff
@@ -143,22 +131,35 @@ public Boolean dispatchKeyEvent (KeyEvent keyEvent){
         return this.view.dispatchKeyEvent(keyEvent);
     }
 
+    public void setDragbar(int viewId){
+        View dragBar = view.findViewById(viewId);
+        if(dragBar == null){
+            return;
+        }
+
+        hasDragBar = true;
+        dragBar.setOnTouchListener((MotionEvent event, View v) -> {
+            onDragBarMovement(event);
+        });
+    }
+
+    private void onDragBarMovement(MotionEvent motionEvent){
+        if ((motionEvent.getAction() == MotionEvent.ACTION_DOWN)) {
+            isDragging = true;
+        } else if (isDragging && (motionEvent.getAction() == MotionEvent.ACTION_UP)) {
+            isDragging = false;
+        }
+    }
+
     public Boolean dispatchTouchEvent (MotionEvent motionEvent) {
-        //Logger.info("touchEvent DragBar "+point.toString());
         if (hasDragBar) {
-            dragBarCollisionBox.initTopLeft(position, dragBarDimensions);
-
-            Vector2Df mousePos = motionEvent.getPosition();
-            if ((motionEvent.getAction() == MotionEvent.ACTION_DOWN) && dragBarCollisionBox.pointCollisionTest(motionEvent.getPosition()) && isDragging == false) {
-                isDragging = true;
-                dragBarClickOffset = Vector2Df.sub(mousePos, this.position);
-                return true;
-            } else if (isDragging && (motionEvent.getAction() == MotionEvent.ACTION_UP)) {
-                isDragging = false;
-                return true;
-            } else if (isDragging && (motionEvent.getAction() == MotionEvent.ACTION_MOVE)) {
-                position = mousePos.sub(dragBarClickOffset);
-                return true;
+            if(isDragging && dragBarClickOffset == null){
+                dragBarClickOffset = Vector2Df.sub(motionEvent.getPosition(), this.position);
+            }
+            else if(isDragging){
+                position = motionEvent.getPosition().sub(dragBarClickOffset);
+            } else {
```

Comment:
you have `else if...` with bracket on the line above and `} else...` in the same if else statement xD